port/unix/omrsysinfo.c

```diff
@@ -3627,25 +3722,26 @@ omrsysinfo_cgroup_is_system_available(struct OMRPortLibrary *portLibrary)
 	int32_t rc = OMRPORT_ERROR_SYSINFO_CGROUP_UNSUPPORTED_PLATFORM;
 
 	if (NULL == PPG_cgroupEntryList) {
-		struct statfs buf = {0};
+		if (isCgroupV1Available(portLibrary)) {
+			BOOLEAN inDocker = FALSE;
 
-		/* If tmpfs is mounted on /sys/fs/cgroup, then it indicates cgroup v1 system is available */
-		rc = statfs(OMR_CGROUP_V1_MOUNT_POINT, &buf);
-		if (0 != rc) {
-			rc = portLibrary->error_set_last_error(portLibrary, errno, OMRPORT_ERROR_SYSINFO_SYS_FS_CGROUP_STATFS_FAILED);
-			goto _end;
-		} else if (TMPFS_MAGIC != buf.f_type) {
-			rc = portLibrary->error_set_last_error_with_message_format(portLibrary, OMRPORT_ERROR_SYSINFO_SYS_FS_CGROUP_TMPFS_NOT_MOUNTED, "tmpfs is not mounted on " OMR_CGROUP_V1_MOUNT_POINT);
-			goto _end;
-		}
+			rc = isRunningInDocker(portLibrary, &inDocker);
+			if (0 != rc) {
+				goto _end;
+			}
+			omrthread_monitor_enter(cgroupEntryListMonitor);
+			if (NULL == PPG_cgroupEntryList) {
+				BOOLEAN useRootCgroup = FALSE;
```

Comment:
Is `useRootCgroup` required or can the code pass `inDocker` instead?

Reply:
We can use `inDocker` instead.